test/gtest/common/test.h

```diff
@@ -88,6 +90,17 @@ class test_base {
 
     virtual void test_body() = 0;
 
+    template<typename T>
+    void wait_for_value(volatile T *var, T value,
+                        double timeout = DEFAULT_TIMEOUT_SEC) const
+    {
+        ucs_time_t deadline = ucs_get_time() + (ucs_time_from_sec(timeout) *
+                                                ucs::test_time_multiplier());
+        while ((ucs_get_time() < deadline) && (*var != value)) {
+            sched_yield();
+        }
+    }
```

Comment:
I counted 3 exiting reincarnations on exactly this function in UCX, and now this is the 4th.
```
ucp_test::wait_for_value
2 overloads in uct_test::wait_for_value
```
Apart from that there are multiple variations of the same function with slightly different names/behavior:
```
wait_progress
wait_for_rx_sn
many more...
```
Maybe we need to generalize it so that we don't have dozens of the similar functions?
Something like:
```C++
    template<typename Cond, typename Wait>
    void wait_for(Cond cond, Wait wait,
                        double timeout = DEFAULT_TIMEOUT_SEC) const
    {
        ucs_time_t deadline = ucs_get_time() + (ucs_time_from_sec(timeout) *
                                                ucs::test_time_multiplier());
        while ((ucs_get_time() < deadline) && (!cond())) {
            wait();
        }
    }

    template<typename T, Wait>
    void wait_for_value(volatile T *var, T value, Wait wait,
                        double timeout = DEFAULT_TIMEOUT_SEC) const
    {
        wait_for([&](){ return (*var == value); }, wait, timeout);
    }

Usage:
wait_for_value(..., sched_yield);

We can have overloads/default values for different wait strategies, or maybe leave it explicit
```

Reply:
fixed